From 22090341358ed2fb0289bc979e7946425a5a5590 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 23 Oct 2024 10:51:09 +0200
Subject: [PATCH 1340/1695] plat-stm32mp2: conf: disable calibration feature if
 M33 is TDCID

Clock Calibration feature is activated by default in stm32mp2 platform,
and disabled if M33 is TDCID.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 639ea2fd58fae3bdcf3c216f24f0a5a3f6ab7778 plat-stm32mp2: conf: disable calibration feature if M33 is TDCID]
Change-Id: I225515e409d4fdca747211959954daa612e9a660
Signed-off-by: Ludovic Barre <ludovic.barre@foss.st.com>
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/412829
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk      | 2 ++
 core/drivers/clk/clk-stm32mp21.c         | 3 +++
 core/drivers/clk/clk-stm32mp25.c         | 5 ++---
 core/drivers/clk/sub.mk                  | 5 +++--
 core/include/drivers/stm32mp2_rcc_util.h | 8 ++++++++
 5 files changed, 18 insertions(+), 5 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index b67158e26..339e3a991 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -97,6 +97,7 @@ $(call force,CFG_SCMI_SERVER_RESET_CONSUMER,n)
 $(call force,CFG_STM32_BSEC_WRITE,n)
 $(call force,CFG_STM32MP_PROVISIONING,n)
 $(call force,CFG_TA_STM32MP_NVMEM,n)
+$(call force,CFG_STM32MP2_CLK_CAL,n)
 endif
 
 $(call force,CFG_ARM_GIC_PM,y)
@@ -120,6 +121,7 @@ $(call force,CFG_STM32_HSE_MONITORING,y)
 $(call force,CFG_STM32_PWR,y)
 $(call force,CFG_STM32_PWR_REGUL,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
+CFG_STM32MP2_CLK_CAL ?= y
 CFG_STM32MP_REMOTEPROC ?= y
 $(call force,CFG_WITH_ARM_TRUSTED_FW,y)
 $(call force,CFG_WITH_LPAE,y)
diff --git a/core/drivers/clk/clk-stm32mp21.c b/core/drivers/clk/clk-stm32mp21.c
index 26fdaeaa3..50f9af650 100644
--- a/core/drivers/clk/clk-stm32mp21.c
+++ b/core/drivers/clk/clk-stm32mp21.c
@@ -4682,6 +4682,9 @@ static TEE_Result stm32mp2_clk_probe(const void *fdt, int node,
 	if (IS_ENABLED(CFG_STM32_CLK_DEBUG))
 		clk_print_tree();
 
+	if (clk_stm32_init_calib(fdt, node))
+		panic("Calibration error init");
+
 	register_pm_core_service_cb(stm32_rcc_pm, NULL, "stm32-rcc");
 
 	return TEE_SUCCESS;
diff --git a/core/drivers/clk/clk-stm32mp25.c b/core/drivers/clk/clk-stm32mp25.c
index 57056181a..fa94e0c58 100644
--- a/core/drivers/clk/clk-stm32mp25.c
+++ b/core/drivers/clk/clk-stm32mp25.c
@@ -5071,9 +5071,8 @@ static TEE_Result stm32mp2_clk_probe(const void *fdt, int node,
 	if (IS_ENABLED(CFG_STM32_CLK_DEBUG))
 		clk_print_tree();
 
-	res = clk_stm32_init_calib(fdt, node);
-	if (res)
-		EMSG("Calibration error init");
+	if (clk_stm32_init_calib(fdt, node))
+		panic("Calibration error init");
 
 	register_pm_core_service_cb(stm32_rcc_pm, NULL, "stm32-rcc");
 
diff --git a/core/drivers/clk/sub.mk b/core/drivers/clk/sub.mk
index d2e988e5f..8b0f5e053 100644
--- a/core/drivers/clk/sub.mk
+++ b/core/drivers/clk/sub.mk
@@ -4,7 +4,8 @@ srcs-$(CFG_DRIVERS_CLK_FIXED) += fixed_clk.c
 srcs-$(CFG_STM32MP_CLK_CORE) += clk-stm32-core.c
 srcs-$(CFG_STM32MP13_CLK) += clk-stm32mp13.c clk-stm32-rcc.c
 srcs-$(CFG_STM32MP15_CLK) += clk-stm32mp15.c clk-stm32-rcc.c
-srcs-$(CFG_STM32MP21_CLK) += clk-stm32mp21.c clk-stm32-rcc.c clk-stm32mp2-cal.c
-srcs-$(CFG_STM32MP25_CLK) += clk-stm32mp25.c clk-stm32-rcc.c clk-stm32mp2-cal.c
+srcs-$(CFG_STM32MP21_CLK) += clk-stm32mp21.c clk-stm32-rcc.c
+srcs-$(CFG_STM32MP25_CLK) += clk-stm32mp25.c clk-stm32-rcc.c
+srcs-$(CFG_STM32MP2_CLK_CAL) += clk-stm32mp2-cal.c
 
 subdirs-$(CFG_DRIVERS_SAM_CLK) += sam
diff --git a/core/include/drivers/stm32mp2_rcc_util.h b/core/include/drivers/stm32mp2_rcc_util.h
index 5d2534722..153c9f0da 100644
--- a/core/include/drivers/stm32mp2_rcc_util.h
+++ b/core/include/drivers/stm32mp2_rcc_util.h
@@ -23,6 +23,14 @@ unsigned long clk_stm32_clock_frequency_calculator(uint32_t ckintsel);
 unsigned long clk_stm32_read_hsi_frequency(void);
 unsigned long clk_stm32_read_msi_frequency(void);
 
+#if defined(CFG_STM32MP2_CLK_CAL)
 TEE_Result clk_stm32_init_calib(const void *fdt, int node);
+#else /* defined(CFG_STM32MP2_CLK_CAL) */
+static inline TEE_Result clk_stm32_init_calib(const void *fdt __unused,
+					      int node __unused)
+{
+	return TEE_SUCCESS;
+}
+#endif /* defined(CFG_STM32MP2_CLK_CAL) */
 
 #endif /*__DRIVERS_STM32MP2_RCC_UTIL_H__*/
-- 
2.47.2

